connectors/kafka/src/main/java/forklift/message/KafkaMessage.java

```diff
@@ -0,0 +1,67 @@
+package forklift.message;
+
+import forklift.connectors.ConnectorException;
+import forklift.connectors.ForkliftMessage;
+import forklift.consumer.parser.KeyValueParser;
+import forklift.controller.KafkaController;
+import forklift.producers.KafkaForkliftProducer;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+public class KafkaMessage extends ForkliftMessage {
+    private final KafkaController controller;
+    private final ConsumerRecord<?, ?> consumerRecord;
+
+    public KafkaMessage(KafkaController controller, ConsumerRecord<?, ?> consumerRecord) {
+        this.controller = controller;
+        this.consumerRecord = consumerRecord;
+        parseRecord();
+    }
+
+    public ConsumerRecord<?, ?> getConsumerRecord() {
+        return this.consumerRecord;
+    }
+
+    @Override
+    public boolean acknowledge() throws ConnectorException {
+        try {
+            return controller.acknowledge(consumerRecord);
+        } catch (InterruptedException e) {
+            throw new ConnectorException("Error acknowledging message");
+        }
+    }
+
+    @Override
+    public String getId() {
```

Comment:
Not really.  You can't set an id on a kafka message, we would need to wire it into the properties or something.  I think the responsibilities of the ForkliftMessage class are worth a discussion at some point.